src/Core/Managed/Shared/Extensibility/MetricAggregator.cs

```diff
@@ -0,0 +1,172 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+    /// <summary>
+    /// Represents aggregator for a single time series of a given metric.
+    /// </summary>
+    public class MetricAggregator : IEquatable<MetricAggregator>
```

Comment:
Are we implementing `IEquatable` in `MetricAggregator` only for the sake of the `MetricAggregatorManager.GetStatisticsAggregator`? It's a lot more code in the `MetricAggregator` and harder to understand in the `MetricAggregatorManager`. Why not simply use `MetricAggregator.aggregatorId` directly?